Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,572 @@
+        if (CFStringGetLength(name) == 0) {
+            TraceError(TAG, L"Failed to convert the localized string to wide string.");
+            return fontFamilyNames;
+        }
+
+        CFArrayAppendValue(fontFamilyNames, name);
+    }
+
+    return fontFamilyNames;
+}
+
+/**
+ * Helper method to retrieve names of individual fonts under a font family.
+ */
+CFArrayRef _DWriteGetFontNamesForFamilyName(CFStringRef familyName) {
+    CFMutableArrayRef fontNames = CFArrayCreateMutable(kCFAllocatorSystemDefault, 0, &kCFTypeArrayCallBacks);
+    CFAutorelease(fontNames);
+
+    ComPtr<IDWriteFactory> dwriteFactory;
+    THROW_IF_FAILED(_DWriteCreateFactoryInstance(&dwriteFactory));
+
+    // Get the system font collection.
+    ComPtr<IDWriteFontCollection> fontCollection;
+    THROW_IF_FAILED(dwriteFactory->GetSystemFontCollection(&fontCollection));
+
+    // Get the font family.
+    size_t index = 0;
+    BOOL exists = false;
+
+    CFIndex familyNameLength = CFStringGetLength(familyName);
+    std::vector<UniChar> unicharFamilyName(familyNameLength + 1);
+    CFStringGetCharacters(familyName, CFRangeMake(0, familyNameLength), unicharFamilyName.data());
+
+    THROW_IF_FAILED(fontCollection->FindFamilyName(reinterpret_cast<wchar_t*>(unicharFamilyName.data()), &index, &exists));
+    if (!exists) {
+        TraceError(TAG, L"Failed to find the font family name.");
+        return fontNames;
+    }
+
+    ComPtr<IDWriteFontFamily> fontFamily;
+    THROW_IF_FAILED(fontCollection->GetFontFamily(index, &fontFamily));
+
+    ComPtr<IDWriteFontList> fontList;
+    THROW_IF_FAILED(
+        fontFamily->GetMatchingFonts(DWRITE_FONT_WEIGHT_THIN, DWRITE_FONT_STRETCH_UNDEFINED, DWRITE_FONT_STYLE_NORMAL, &fontList));
+
+    size_t count = fontList->GetFontCount();
+
+    for (size_t i = 0; i < count; i++) {
+        ComPtr<IDWriteFont> font;
+        THROW_IF_FAILED(fontList->GetFont(i, &font));
+
+        ComPtr<IDWriteLocalizedStrings> fullName;
+        BOOL exist = FALSE;
+        THROW_IF_FAILED(font->GetInformationalStrings(DWRITE_INFORMATIONAL_STRING_FULL_NAME, &fullName, &exist));
+
+        if (exist) {
+            CFStringRef name = _CFStringFromLocalizedString(fullName.Get());
+            if (CFStringGetLength(name) == 0) {
+                TraceError(TAG, L"Failed to convert the localized string to wide string.");
+                return fontNames;
+            }
+
+            CFArrayAppendValue(fontNames, name);
+        }
+    }
+
+    return fontNames;
+}
+
+/**
+ * Helper method that maps a font name to the name of its family.
+ *
+ * Note: This function currently uses a cache, meaning that fonts installed during runtime will not be reflected
+ */
+CFStringRef _DWriteGetFamilyNameForFontName(CFStringRef fontName) {
+    static CFDictionaryRef fontToFamilyMap = nullptr;
+
+    CFLocaleRef locale = CFLocaleCopyCurrent();
+    CFAutorelease(locale);
+
+    static dispatch_once_t initOnce = 0;
```

Comment:
nope! if the variable is static the lambda assigning it will only be called once, at least it better be or our activation scheme is terribly flawed...